addons/share-through-mystuff/userscript.js

```diff
@@ -2,26 +2,27 @@ export default async function ({ addon }) {
   const shareFunction = document.createElement("a");
   shareFunction.classList.add("media-share");
   shareFunction.dataset.control = "share";
-  shareFunction.href = "#";
 
   const shareButton = document.createElement("a");
-  async function shareConfirmation(event) {
-    let confirmation = await addon.tab.confirm("Share this project?", "Are you sure you want to share this project?");
-    if (confirmation) event.target.parentElement.querySelector(".media-share").click();
-  }
-  shareButton.href = "#";
   shareButton.classList.add("sa-share-button");
   shareButton.innerText = "Share";
   addon.tab.displayNoneWhileDisabled(shareButton, {
     display: "block",
   });
+  async function shareConfirmation(event) {
+    let confirmation = await addon.tab.confirm("Share this project?", "Are you sure you want to share this project?");
+    if (confirmation) event.target.parentElement.querySelector(".media-share").click();
+  }
 
   while (true) {
     const project = await addon.tab.waitForElement("div.media-item-content.not-shared", {
       markAsSeen: true,
     });
-    project.querySelector(".media-action div").appendChild(shareFunction.cloneNode());
+    let localShareFunction = shareFunction.cloneNode();
     let localShare = shareButton.cloneNode(true);
+    localShareFunction.href = `${location.hash}`;
+    localShare.href = localShareFunction.href;
```

Comment:
This seems to cause the page to reload before the confirmation pops up.

Reply:
`location.hash` has a `#` at the beginning of the string it returns (as long as there is one to return). A `#` at the beginning of a `href` makes it _not_ reload the page. So I a don't think this is the problem.

Reply:
There are two separate bugs: the first is when the all projects tab is selected clicking the share link reloads the page instantly without sharing the project. The second is when the unshared projects tab is selected sharing works as expected and doesn't reload the page, but after sharing the project all the project on that tab disappear untul I reload it.

Reply:
Try sharing a project with the share project tab open.

Reply:
Also I think I fixed the other bug.